org.openhab.binding.zigbee/src/main/java/org/openhab/binding/zigbee/internal/converter/config/ZclReportingConfig.java

```diff
@@ -0,0 +1,179 @@
+
+    private static final Integer REPORTING_RANGE_MIN = 1;
+    private static final Integer REPORTING_RANGE_MAX = 86400;
+
+    private static final Integer REPORTING_DEFAULT_MIN = 1;
+    private static final Integer REPORTING_DEFAULT_MAX = 900;
+
+    private static final Integer POLLING_RANGE_MIN = 15;
+    private static final Integer POLLING_RANGE_MAX = 86400;
+    private static final Integer POLLING_DEFAULT = 900;
+
+    private static final String CONFIG_ID = "zigbee_reporting_";
+    private static final String CONFIG_REPORTINGMIN = CONFIG_ID + "min";
+    private static final String CONFIG_REPORTINGMAX = CONFIG_ID + "max";
+    private static final String CONFIG_REPORTINGCHANGE = CONFIG_ID + "change";
+    public static final String CONFIG_POLLING = CONFIG_ID + "polling";
+
+    private int reportingTimeMin = REPORTING_DEFAULT_MIN;
+    private int reportingTimeMax = REPORTING_DEFAULT_MAX;
+    private int reportingChange = 1;
+    private int pollingPeriod = POLLING_DEFAULT;
+
+    // Reporting change configuration. These will need to be set for each channel.
+    private boolean isAnalogue = false;
+    private BigDecimal defChange;
+    private BigDecimal minChange;
+    private BigDecimal maxChange;
+
+    @Override
+    public boolean initialize(ZclCluster cluster) {
+        return true;
+    }
+
+    public void setAnalogue(BigDecimal defChange, BigDecimal minChange, BigDecimal maxChange) {
+        this.isAnalogue = true;
+        this.defChange = defChange;
+        this.minChange = minChange;
+        this.maxChange = maxChange;
+    }
+
+    @Override
+    public List<ConfigDescriptionParameter> getConfiguration() {
+        List<ConfigDescriptionParameter> parameters = new ArrayList<>();
+
+        // Build a list of configuration
+        parameters.add(ConfigDescriptionParameterBuilder.create(CONFIG_REPORTINGMIN, Type.INTEGER)
+                .withLabel("Minimum Reporting Period")
+                .withDescription("The minimum time period in seconds between device state updates")
+                .withDefault(REPORTING_DEFAULT_MIN.toString()).withMinimum(new BigDecimal(REPORTING_RANGE_MIN))
+                .withMaximum(new BigDecimal(REPORTING_RANGE_MAX)).build());
+
+        parameters.add(ConfigDescriptionParameterBuilder.create(CONFIG_REPORTINGMAX, Type.INTEGER)
+                .withLabel("Maximum Reporting Period")
+                .withDescription("The maximum time period in seconds between device state updates")
+                .withDefault(REPORTING_DEFAULT_MAX.toString()).withMinimum(new BigDecimal(REPORTING_RANGE_MIN))
+                .withMaximum(new BigDecimal(REPORTING_RANGE_MAX)).build());
+
+        if (isAnalogue) {
+            parameters.add(ConfigDescriptionParameterBuilder.create(CONFIG_REPORTINGCHANGE, Type.INTEGER)
+                    .withLabel("Report On Change")
+                    .withDescription("The maximum time period between device state updates")
```

Comment:
Should the description rather be something like `The minimum change of the attribute value needed to trigger a device state update`?